### DISCUSSION TITLE
Not seeing expected improvement in throughput of RaftCluster.ReplicateAsync method when cluster minority is inaccessible

### DISCUSSION THREAD

**Discussion by LarsWithCA** [dotnet/dotNext#231] [General]
Hello there!

This plots shows the timing (in ms) of RaftCluster.ReplicateAsync - at the vertical green line 1 node is disconnected (out of a cluster of 6 nodes in total):
(Linux ARM + .NET6 + DotNext.Net.Cluster 4.14.1)
![image](https://github.com/dotnet/dotNext/assets/132266664/834cf408-02d4-4dc9-bfa6-ea7e6d880bdb)

From the change log of DotNext.Net.Cluster 4.15.0:

> Raft performance: improved throughput of IRaftCluster.ReplicateAsync method when cluster minority is not accessible (faulty node). Now the leader waits for replication from majority of nodes instead of all nodes

This made us hope that we would no longer see these kinds of longer timings in case of inaccessible cluster minority. However, we see a pretty similar plot - at the green line 1 node is disconnected (out of a cluster of 6 nodes):
(Linux ARM + .NET8 + DotNext.Net.Cluster 5.3.0)
![image](https://github.com/dotnet/dotNext/assets/132266664/551ce738-23ae-4a9a-b703-375777764236)

Did we have wrong expectations, or are we doing something wrong?

**Comment by sakno** [comment]
I'll check this out.

**Comment by sakno** [comment]
For now, appended log entries are marked as committed when replicated to the majority. `ForceReplicationAsync` remains suspended while receiving responses from all (even unavailable) members. So information about increased throughput is partially true. Will take a look how resume callers earlier.

**Reply by sakno** [reply]
@LarsWithCA , could you please create an issue?

**Reply by LarsWithCA** [reply]
Hi @sakno I created https://github.com/dotnet/dotNext/issues/233

**Comment by LarsWithCA** [comment]
Fixed in version 5.5.0
